File: src-tauri/src/commands/league/league.rs
use flint_core::league::{detect_league_installation, validate_league_path, LeagueInstallation};

#[tauri::command]
pub async fn detect_league() -> Result<LeagueInstallation, String> {
    tracing::info!("Frontend requested League detection");
    
    tokio::task::spawn_blocking(move || {
        detect_league_installation()
    })
    .await
    .map_err(|e| format!("Task failed: {}", e))?
    .map_err(|e| e.to_string())
}

#[tauri::command]
pub async fn validate_league(path: String) -> Result<LeagueInstallation, String> {
    tracing::info!("Frontend requested validation for path: {}", path);
    
    tokio::task::spawn_blocking(move || {
        validate_league_path(&path)
    })
    .await
    .map_err(|e| format!("Task failed: {}", e))?
    .map_err(|e| e.to_string())
}

File: src/lib/api/league.ts
import { invokeCommand } from './core';

export async function detectLeague(): Promise<{ path: string; source: string }> {
    return invokeCommand('detect_league');
}

interface LeagueInstallation {
    path: string;
    game_path: string;
    auto_detected: boolean;
}

export async function validateLeague(path: string): Promise<{ valid: boolean; path: string | null }> {
    try {
        const result = await invokeCommand<LeagueInstallation>('validate_league', { path });
        return { valid: true, path: result.path };
    } catch {
        return { valid: false, path: null };
    }
}
